canister/src/types.rs

```diff
@@ -52,37 +59,82 @@ impl Default for Config {
     }
 }
 
+#[derive(
+    CandidType, Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Copy, Default, PartialOrd, Ord,
+)]
+pub struct Cycles(u128);
+
+impl Cycles {
+    pub const fn new(input: u128) -> Self {
+        Self(input)
+    }
+
+    pub fn get(self) -> u128 {
+        self.0
+    }
+}
+
+impl fmt::Display for Cycles {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0.separate_with_underscores())
+    }
+}
+
+impl Add for Cycles {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Self(self.0.saturating_add(rhs.0))
+    }
+}
+
+impl Mul<u64> for Cycles {
+    type Output = Self;
+
+    fn mul(self, rhs: u64) -> Self {
+        Self(self.0.saturating_mul(rhs as u128))
+    }
+}
+
+impl Mul<usize> for Cycles {
+    type Output = Self;
+
+    fn mul(self, rhs: usize) -> Self {
+        Self(self.0.saturating_mul(rhs as u128))
+    }
+}
+
 #[derive(CandidType, Serialize, Deserialize, PartialEq, Eq, Debug, Clone, Default)]
 pub struct Fees {
     /// The base fee to charge for all `get_utxos` requests.
-    pub get_utxos_base: u128,
+    pub get_utxos_base: Cycles,
```

Comment:
I'm not sure this change is fine from a backwards compatibility point of view. We'll need to double check whether we can successfully upgrade the canister to this version.

Reply:
That makes sense. I created another PR, simplified version with only adding underscores and no changing types.

https://github.com/dfinity/bitcoin-canister/pull/119